packages/contractkit/src/utils/tx-result.ts

```diff
@@ -0,0 +1,59 @@
+import debugFactory from 'debug'
+import { ExternalPromise } from 'src/utils/external-promise'
+import PromiEvent from 'web3/promiEvent'
+import { TransactionReceipt } from 'web3/types'
+
+const debug = debugFactory('contractkit:txresult')
+
+export function toTxResult(pe: PromiEvent<any>) {
+  return new TransactionResult(pe)
+}
+
+export class TransactionResult {
+  private hashFuture = new ExternalPromise<string>()
+  private receiptFuture = new ExternalPromise<TransactionReceipt>()
+  private confirmationsFuture = new ExternalPromise<boolean>()
+
+  constructor(pe: PromiEvent<any>) {
+    pe.on('transactionHash', (hash: string) => {
+      debug('hash: %s', hash)
+      this.hashFuture.resolve(hash)
+    })
+      .on('receipt', (receipt: TransactionReceipt) => {
+        debug('receipt: %O', receipt)
+        this.receiptFuture.resolve(receipt)
+      })
+      .on('confirmation', (confirmationNumber: number) => {
```

Comment:
If we have finality within 1 block, do we need this? Or is it sufficient to be able to wait for the receipt?

Reply:
good question...
this was ported from old contractkit..

i don't know if we want this or not

Reply:
I'll remove this